resource/policies/base/match_op.h

```diff
@@ -0,0 +1,34 @@
+#ifndef MATCH_OP_H
+#define MATCH_OP_H
+
+
+enum match_op_t { MATCH_UNKNOWN,
+                  MATCH_ALLOCATE,
+                  MATCH_ALLOCATE_W_SATISFIABILITY,
+                  MATCH_ALLOCATE_ORELSE_RESERVE,
+                  MATCH_SATISFIABILITY };
+
+static const char *match_op_to_string (enum match_op_t match_op) {
+    switch (match_op) {
+        case MATCH_ALLOCATE:                  return "allocate";
+        case MATCH_ALLOCATE_ORELSE_RESERVE:   return "allocate_orelse_reserve";
+        case MATCH_ALLOCATE_W_SATISFIABILITY: return "allocate_with_satisfiability";
+        case MATCH_SATISFIABILITY:            return "satisfiability";
+        default: return "error";
+    }
+}
+
+static bool match_op_valid (enum match_op_t match_op) {
```

Comment:
No need to specify `enum` type here:
```suggestion
static bool match_op_valid (match_op_t match_op) {
```